src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -632,24 +632,38 @@ private void getContainerInfoBuilder(Offer offer, Request request, String slaveN
         return commandBuilder;
   }
 
-  String generateJenkinsCommand2Run(int jvmMem,String jvmArgString,String jnlpArgString,String slaveName) {
+  String generateJenkinsCommand2Run(int jvmMem, String jvmArgString, String jnlpArgString, String slaveName, List<MesosSlaveInfo.Command> additionalCommands) {
 
-      return String.format(SLAVE_COMMAND_FORMAT,
-              jvmMem,
-              jvmArgString,
-              jnlpArgString,
-              getJnlpSecret(slaveName),
-              getJnlpUrl(slaveName));
+    String slaveCmd = String.format(SLAVE_COMMAND_FORMAT,
+            jvmMem,
+            jvmArgString,
+            jnlpArgString,
+            getJnlpSecret(slaveName),
+            getJnlpUrl(slaveName));
+
+    StringBuilder commandStringBuilder = new StringBuilder();
+
+    if (additionalCommands != null && !additionalCommands.isEmpty()) {
+      for (MesosSlaveInfo.Command additionalCommand : additionalCommands) {
+        commandStringBuilder.append(additionalCommand.getValue() + " && ");
```

Comment:
hmm. i think it won't be obvious to users that the commands are tied by "&&". how about having user input just one shell command instead of a list? the user can easily create a composite shell command (using && or ||) himself.